test-integration/operator-tests/src/test/java/org/infinispan/operator/CustomLibsIT.java

```diff
@@ -0,0 +1,107 @@
+package org.infinispan.operator;
+
+import cz.xtf.core.config.XTFConfig;
+import cz.xtf.core.openshift.OpenShift;
+import cz.xtf.core.openshift.OpenShiftWaiters;
+import cz.xtf.core.openshift.OpenShifts;
+import cz.xtf.junit5.annotations.CleanBeforeAll;
+import io.fabric8.kubernetes.api.model.*;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.maven.it.VerificationException;
+import org.apache.maven.it.Verifier;
+import org.assertj.core.api.Assertions;
+import org.infinispan.Infinispan;
+import org.infinispan.Infinispans;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Paths;
+import java.util.Collections;
+
+@Slf4j
+@CleanBeforeAll
+public class CustomLibsIT {
+    private static final OpenShift openShift = OpenShifts.master();
+    private static final OpenShiftWaiters openShiftWaiters = OpenShiftWaiters.get(openShift, () -> false);
+    private static final Infinispan infinispan = Infinispans.customLibs();
+
+    @BeforeAll
+    static void prepare() throws IOException {
+        preparePVC();
+        preparePod();
+
+        openShiftWaiters.areExactlyNPodsRunning(1).waitFor();
+
+        buildLibs();
+        uploadLibs();
+
+        infinispan.deploy();
+        infinispan.waitFor();
+    }
+
+    @Test
+    void test() {
+        Pod pod = openShift.getPod("custom-libs-0");
+        String logs = openShift.getPodLog(pod);
+
+        Assertions.assertThat(logs).contains("Loaded extension 'basic-filter-factoy'");
+    }
+
+    private static void preparePVC() {
+        PersistentVolumeClaimBuilder pvc = new PersistentVolumeClaimBuilder();
+        pvc.withNewMetadata().withName("infinispan-libs").endMetadata();
+        pvc.withNewSpec()
+                .withAccessModes("ReadWriteOnce")
+                .withNewResources().withRequests(Collections.singletonMap("storage", new Quantity("100Mi")))
+                .endResources().endSpec();
+
+        openShift.createPersistentVolumeClaim(pvc.build());
+    }
+
+    private static void preparePod() {
+        String serverImage = XTFConfig.get("org.infinispan.image.server", "quay.io/infinispan/server:latest");
+
+        VolumeBuilder vb = new VolumeBuilder();
+        vb.withName("lib-pv-storage").withNewPersistentVolumeClaim("infinispan-libs", false);
+
+        VolumeMountBuilder vmb = new VolumeMountBuilder();
+        vmb.withMountPath("/tmp/libs");
+        vmb.withName("lib-pv-storage");
+
+        ContainerBuilder cb = new ContainerBuilder();
+        cb.withName("lib-pv-container")
+                .withImage(serverImage)
+                .withVolumeMounts(vmb.build());
+
+        PodBuilder pod = new PodBuilder();
```

Comment:
I tried to run this code on my OCP but it hangs with
```
[INFO] Running org.infinispan.operator.CustomLibsIT
[19:17:04.958] INFO - Waiting up to 20 seconds. Reason: Cleaning project.
[19:17:07.096] INFO - Waiting up to 3 minutes. Reason: Waiting for exactly 1 pods to be running.
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 215.183 s <<< FAILURE! - in org.infinispan.operator.CustomLibsIT
[ERROR] org.infinispan.operator.CustomLibsIT  Time elapsed: 215.181 s  <<< ERROR!
cz.xtf.core.waiting.WaiterException
```
however Pod was created successfully
```
✔ /u01/git/infinispan/infinispan-operator/test-integration [pr-1011 L|✚ 3] 
19:20 $ oc get pods
NAME              READY   STATUS    RESTARTS   AGE
infinispan-libs   1/1     Running   0          8m36s
✔ /u01/git/infinispan/infinispan-operator/test-integration [pr-1011 L|✚ 3]
```

Reply:
I haven't tried it yet. The design is that Operator runs in different one as it's technically a static resource. You can try adding labels `xtf.cz/keep` to all the Operator resources so they are not wiped out. But if you are going to run the Operator in the same namespace the waiter expecting only one pod running in the namespace will fail and would have to be updated to be more specif

[1] https://github.com/xtf-cz/xtf/blob/master/core/src/main/java/cz/xtf/core/openshift/OpenShift.java#L95

Reply:
Oh, good, thanks @Crumby , will try to execute and probably add this options to local operator deployment script